### PR TITLE
feat(swap): add tx ref to broadcast

### PR DESCRIPTION
**Pull request by zoheb391**
closes WEB-987

### CODE REVIEW

File: packages/swap/src/event-handlers/networkBroadcastSuccess.ts [hunk 1]
```diff
@@ -1,24 +1,40 @@
...
-import { unsignedInteger } from '@/shared/parsers';
+import { hexString, unsignedInteger } from '@/shared/parsers';
 import type { EventHandlerArgs } from './index';
 
-const eventArgs = z.object({
+export const networkBroadcastSuccessArgs = z.object({
   broadcastId: unsignedInteger,
+  transactionRef: z // v130+
+    .union([
+      z.object({ blockNumber: z.number(), extrinsicIndex: z.number() }),
```
Comment (acdibble) [inline packages/swap/src/event-handlers/networkBroadcastSuccess.ts:10]:
```suggestion
      z.object({ blockNumber: z.number(), extrinsicIndex: z.number() })
        .transform(({ blockNumber, extrinsicIndex }) => `${blockNumber}-${extrinsicIndex}`),
```

File: packages/swap/src/event-handlers/networkBroadcastSuccess.ts [hunk 2]
```diff
@@ -1,24 +1,40 @@
...
+    const args = networkBroadcastSuccessArgs.parse(event.args);
+
+    let txRef;
+    if (args.transactionRef) {
+      if (typeof args.transactionRef === 'string') {
+        txRef = args.transactionRef;
+      } else {
+        txRef = `${args.transactionRef.blockNumber}-${args.transactionRef.extrinsicIndex}`;
+      }
+    }
```
Comment (acdibble) [inline packages/swap/src/event-handlers/networkBroadcastSuccess.ts:29]:
```suggestion
```

File: packages/swap/src/event-handlers/networkBroadcastSuccess.ts [hunk 3]
```diff
@@ -1,24 +1,40 @@
...
 
     // use updateMany to skip update if broadcast does not include any swap
     await prisma.broadcast.updateMany({
-      where: { chain, nativeId: broadcastId },
+      where: { chain, nativeId: args.broadcastId },
       data: {
         succeededAt: new Date(block.timestamp),
         succeededBlockIndex: `${block.height}-${event.indexInBlock}`,
+        transactionRef: txRef,
```
Comment (acdibble) [inline packages/swap/src/event-handlers/networkBroadcastSuccess.ts:37]:
```suggestion
        transactionRef: args.transactionRef,
```

**Comment by zoheb391** [comment]
We also need to expose this on the swapping backend/sdk. Do we have any preferences for what we want to call this field?

I'm thinking `outboundTxRef`
or 
`outboundTxLink` with the full block explorer link

**Review by acdibble — APPROVED** [review]

File: packages/swap/src/event-handlers/networkBroadcastSuccess.ts (hunk 1 above)
Comment (zoheb391) [reply packages/swap/src/event-handlers/networkBroadcastSuccess.ts:10]: duh ofcourse

**Review by niklasnatter — APPROVED** [review]
i would call the property on the status `broadcastTransactionRef` because we have most of the properties prefixed with the state chain term. and i think providing the ref (not an explorer link) is preferable bc its easier to create an explorer link from a ref than the other way

**Pull request merged**